baseapp/baseapp.go

```diff
@@ -242,15 +285,27 @@ func (app *BaseApp) SetOption(req abci.RequestSetOption) (res abci.ResponseSetOp
 }
 
 // Implements ABCI
-// InitChain runs the initialization logic directly on the CommitMultiStore and commits it.
+// InitChain runs the initialization logic directly on the CommitMultiStore.
 func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitChain) {
+
+	// Stash the consensus params in the cms main store and memoize.
+	if req.ConsensusParams != nil {
+		app.setConsensusParams(req.ConsensusParams)
+		app.storeConsensusParams(req.ConsensusParams)
```

Comment:
Why do we need to store these in the store? Won't they always be sent on `InitChain` to the ABCI application (so we can just store them in the baseapp struct)?

Reply:
From @jaekwon 
```
two scenarios… (1) upon genesis, it won't exist… so initFromStore will not load anything… but later initChainer will give it to the app, and baseapp can store it then.

(2) upon restart, app initialization calls LoadLatestVersion > initFromStore > loads the paramsKeeper's consensus-state subspace.

and from (1) (during initChainer) and (2) (during initFromStore) it gets stashed on a baseapp.consensusParams via I guess baseapp.SetConsensusParams().
```

Reply:
aka if you restart from a running chain you need to reload the ConsensusParams pointer and InitChain will not be called - hence it needs to be stored

Reply:
I see, thanks!